app/uk/gov/hmrc/agentsexternalstubs/controllers/DesStubController.scala

```diff
@@ -426,6 +426,49 @@ class DesStubController @Inject()(
     }(SessionRecordNotFound)
   }
 
+  def getCgtKnownFacts(regime: String, id: String): Action[AnyContent] = Action.async { implicit request =>
+    withCurrentSession { session =>
+      RegexPatterns
+        .validCgtRef(id)
+        .fold(
+          _ => badRequestF("INVALID_IDVALUE", "Submission has not passed validation. Invalid parameter idValue."),
+          _ =>
+            if (regime != "CGT") {
+              badRequestF("INVALID_REGIME", "Submission has not passed validation. Invalid parameter regimeValue.")
+            } else {
+              usersService
+                .findByPrincipalEnrolmentKey(
+                  EnrolmentKey("HMRC-CGT-PD", Seq(Identifier("CGTPDRef", id))),
+                  session.planetId)
+                .map {
+                  case Some(record) =>
+                    val tpd = record.affinityGroup match {
+                      case Some("Individual") =>
+                        TypeOfPersonDetails(
+                          "Individual",
+                          Left(IndividualName(record.firstName.getOrElse(""), record.lastName.getOrElse(""))))
+                      case _ => TypeOfPersonDetails("Trustee", Right(OrganisationName(record.name.getOrElse(""))))
```

Comment:
This code makes several assumptions - please could you document them?  Are the options really only "Individual" and "Trustee"?  Can this change, or is it fixed in stone?  Please add a comment as to why it can be only one or the other, or add an explicit case for "Trustee" with fallback to not found (or some other error).

Reply:
it's as per spec I have implemented here... they never change should be either "Individual" or "Trustee"

Reply:
OK